### PR TITLE
typings: Add node 10+ Container.connect and .listen options

### PR DESCRIPTION
**Pull request by kohtala**
Some node library server and client options were missing in typings.
Use the definitions from @types/node to make them complete.
Typings did not validate for consistent TLS vs TCP vs IPC options.

The [tls.Server](https://nodejs.org/api/tls.html#tlscreateserveroptions-secureconnectionlistener) can take options, but they were not passed to it.

Had old @types for node and mocha and old typescript, as tests now run from
node 10 up. Some tests did not parse with new typescript and had errors.

Protocol and connection errors did not capture stack traces.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -320,6 +323,28 @@ object that may contain any of the following fields:
   * autosettle - Whether received messages should be automatically
     settled once the remote settles them. Defaults to true.
 
+And attach frame fields:
+
+  * role
```
Comment (grs) [inline README.md:328]: The role cannot be set.

File: README.md [hunk 2]
```diff
@@ -336,20 +361,13 @@ The argument to this method can either be a simple string indicating
...
-    * expiry_policy
-    * durable
-  * source - The source of a sending link is the local identifier. It
-    is usually not needed, but can be set if it is,
   * name - The name of the link. This should be unique for the
     container. If not specified a unqiue name is generated.
   * autosettle - Whether sent messages should be automatically
     settled once the peer settles them. Defaults to true.
 
+And attach frame fields as for `open_receiver`.
```
Comment (grs) [inline README.md:369]: Source and target are used differently for senders and receivers, so should be pulled out of the common set or the descriptions are wrong for senders.

File: package.json [hunk 3]
```diff
@@ -14,20 +14,21 @@
         "tls": false
     },
     "dependencies": {
+        "browserify": "^17.0.0",
```
Comment (grs) [inline package.json:17]: This is a devDependency not a runtime dependency

File: README.md (hunk 1 above)
Comment (kohtala) [reply README.md:328]: Ah. True. I copied them from code and dropped some, but this was left.

File: README.md (hunk 2 above)
Comment (kohtala) [reply README.md:369]: True.

File: package.json (hunk 3 above)
Comment (kohtala) [reply package.json:17]: Yes. Missed that it had got there. I found in command history I had run `npm upgrade browserify`. It did not have version before, so not sure why I had done that. If npm warned it required `--dev` option, I missed it too.

I have an old draft PR that contained a debug upgrade commit. I'll toss that in as well. No reason to keep old debug for node 4 support.

File: README.md (hunk 1 above)
Comment (kohtala) [reply README.md:328]: The incomplete_unsettled and initial_delivery_count are also internal.

**Comment by kohtala** [comment]
Thanks for comments.

**Comment by grs** [comment]
Thanks again @kohtala, much appreciated!

**Comment by grs** [comment]
I've published 2.0.8 without this commit, and then republished as 3.0.0

**Pull request merged**